include/drivers/interrupt_controller/gic.h

```diff
@@ -7,7 +7,7 @@
 #ifndef ZEPHYR_INCLUDE_DRIVERS_GIC_H_
 #define ZEPHYR_INCLUDE_DRIVERS_GIC_H_
 
-#include <arch/cpu.h>
+#include <kernel.h>
```

Comment:
hm, why kernel inclusion is needed here?
I wonder if this could go in the source file (and only if needed actually)

Reply:
Changed to include `device.h` instead